pkg/client/typed/dynamic/client.go

```diff
@@ -91,6 +98,13 @@ type ResourceClient struct {
 	ns       string
 }
 
+// SubresourceClient is an API interface to a specific subresource of a resource under a
+// dynamic client.
+type SubresourceClient struct {
+	*ResourceClient
```

Comment:
I don't think this should be anonymous inclusion.  Doing it like that means that you can pick up all sorts of methods that don't completely make sense.  Even now, you've got a `Get` method for subresources which aren't even allowed to have names.  It's creating an API with "don't push *that* button" kind of dangers.